src/track/keyutils.cpp

```diff
@@ -298,8 +353,8 @@ ChromaticKey KeyUtils::guessKeyFromText(const QString& text) {
 
         int openKeyNumber = lancelotNumberToOpenKeyNumber(lancelotNumber);
 
-        bool major = lancelotKeyMatcher.cap(2)
-                .compare("b", Qt::CaseInsensitive) == 0;
+        const QChar lancelotScaleMode = lancelotKeyMatcher.cap(2).at(0);
+        bool major = (QString("BILM").indexOf(lancelotScaleMode) != -1);
```

Comment:
The meaning of "BILM" is not immediately clear, can you make this static and put it near the definition of `s_scaleModeIsMajor`?
Also, we loose the case-insensitivy of the previous check is that intentional, if not, we should make the string `"BbIiLlMm"` or some variant.

Here a a couple suggestions on how to implement this line specifically a little differently: 
This would result in much better codegen because the std::string ctor is constexpr.
```suggestion
        bool major = std::string("BILM").find(c.toLatin1()) != std::string::npos;
```
Even better but needs C++23.
```cpp
        bool major = std::string("BILM").contains(c.toLatin1());
```

Reply:
We allow only capital letters here. This is required to distinguish it reliable form the Open Key Notation.